File: src/main/scala/com/example/modid/proxy/IProxy.scala
package com.example.modid.proxy

import net.minecraftforge.fml.common.event.{FMLInitializationEvent, FMLPostInitializationEvent, FMLPreInitializationEvent}

trait IProxy {

  def preInit(event: FMLPreInitializationEvent): Unit = {
  }

  def init(event: FMLInitializationEvent): Unit = {
  }

  def postInit(event: FMLPostInitializationEvent): Unit = {
  }
}

File: src/main/kotlin/com/example/modid/proxy/IProxy.kt
package com.example.modid.proxy

import net.minecraftforge.fml.common.event.FMLInitializationEvent
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent

interface IProxy {
    fun preInit(event: FMLPreInitializationEvent) {
    }

    fun init(event: FMLInitializationEvent) {
    }

    fun postInit(event: FMLPostInitializationEvent) {
    }
}
